### PR TITLE
Adjust bundled deps in KSP2

### PR DESCRIPTION
**Pull request by ting-yuan**
1. Do not `bundle kotlin-stdlib`
2. Do not `bundle symbol-processing-api`
3. Make sure all bundled deps are properly renamed

### CODE REVIEW

File: symbol-processing-aa-embeddable/build.gradle.kts [hunk 1]
```diff
@@ -51,6 +70,44 @@ fun String.replaceWithKsp() =
 
 val DEP_SOURCES_DIR = "$buildDir/source-jar"
 
+val validPaths = prefixesToRelocate.map {
+    it.second.split('.').filter { it.isNotEmpty() }.joinToString("/")
```
Comment (neetopia) [inline symbol-processing-aa-embeddable/build.gradle.kts:74]: when will it be empty?

File: symbol-processing-aa-embeddable/build.gradle.kts [hunk 2]
```diff
@@ -51,6 +70,44 @@ fun String.replaceWithKsp() =
...
+    "org/apache/log4j/package-info.class",
+    "ksp/FirNativeForwardDeclarationGetClassCallChecker.class",
+)
+
+class Trie(paths: List<String>) {
+    class TrieNode(val key: String)
+
+    private val terminals = mutableSetOf<TrieNode>()
+
+    private val m = mutableMapOf<Pair<TrieNode?, String>, TrieNode>().apply {
```
Comment (neetopia) [inline symbol-processing-aa-embeddable/build.gradle.kts:87]: I did not get this, if `m` is the child list of the current trie node, what is this pair used for? Feels like simply putting `String` here should be enough?

File: symbol-processing-aa-embeddable/build.gradle.kts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ting-yuan) [reply symbol-processing-aa-embeddable/build.gradle.kts:74]: zip's specification says that it must be a forward slash.

Comment (ting-yuan) [reply symbol-processing-aa-embeddable/build.gradle.kts:74]: When it's ending with a dot.

File: symbol-processing-aa-embeddable/build.gradle.kts (hunk 2 above)
Comment (ting-yuan) [reply symbol-processing-aa-embeddable/build.gradle.kts:87]: The first element of the pair is parent. `m` is shared among all nodes and it maps (parent, key) to the child of the parent.

**Review by neetopia — APPROVED** [review]

**Pull request merged**